src/components/Calculator.js

```diff
@@ -1,24 +1,50 @@
 import React from 'react';
 import './calculator.scss';
 import KeypadRow from './CalculatorKeypadRow';
+import calculate from '../logic/calculate';
 
 export default class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { result: 0 };
+    this.state = { total: null, next: null, operation: null };
+    this.callCalculate = this.callCalculate.bind(this);
+  }
+
+  callCalculate(value) {
+    const newState = calculate(this.state, value);
+    this.setState(newState);
   }
 
   render() {
-    const { result } = this.state;
+    const { total, next, operation } = this.state;
     return (
       <div className="calculator flex-col-center">
-        <div className="calculator__display">{result}</div>
+        <div className="calculator__display">
+          {operation == null
+            ? total || next || '0'
+            : `${total} ${operation} ${next == null ? ' ' : next}`}
+        </div>
         <div className="calculator__keypad">
-          <KeypadRow values={['AC', '+/-', '%', '÷']} />
-          <KeypadRow values={['7', '8', '9', 'x']} />
-          <KeypadRow values={['4', '5', '6', '-']} />
-          <KeypadRow values={['1', '2', '3', '+']} />
-          <KeypadRow values={['0', '.', '=']} />
+          <KeypadRow
+            values={['AC', '+/-', '%', '÷']}
+            callCalculate={this.callCalculate}
+          />
```

Comment:
- [ ] When I try any number modulo zero the app crashes. Kindly fix the issue, maybe try displaying an error message like when the user tries division by zero.

Reply:
Thanks, That's a really good point